test/project_types/script/forms/create_test.rb

```diff
@@ -46,6 +46,21 @@ def test_name_is_cleaned_when_using_flag
         assert_equal 'name_with_space', form.name
       end
 
+      def test_invalid_name
+        name = 'na/me'
+        CLI::UI::Prompt.expects(:ask).with('Script Name').returns(name)
+
+        io = capture_io do
+          ask(extension_point: 'discount')
+        end
+
+        assert_match(
+          "Invalid script name: replace or remove unsupported characters. Valid "\
+          "characters are numbers, letters, spaces, hyphens, or underscores.",
+          io.join
+        )
```

Comment:
I might be better to have a `FakeContext` in these test classes, so we can do string comparisons with `ctx.message('...')`.  Similarly for texts on line 51, if we could use `ctx.message('...')` instead of `Script Name`.  It will reduce amount of rework once we change these texts in the `messages.rb` file.